### PR TITLE
fix: implement retries for Deno CLI download and additional logging

### PR DESCRIPTION
**Pull request by jackiewmacharia**
🎉 Thanks for sending this pull request! 🎉

This PR adds additional system logging for Edge Functions bundling and adds retries to Deno CLI download.

If retries are triggered for download, logger.system will have a record of retry attempts similar to the below screenshot:

Screenshot 2022-08-16 at 17 14 54

Relates to https://github.com/netlify/pillar-support/issues/22

--------

Please make sure the title is clear and descriptive.

If you are fixing a typo or documentation, please skip these instructions.

Otherwise please fill in the sections below.

**Which problem is this pull request solving?**

Example: I'm always frustrated when [...]

**List other issues or pull requests related to this problem**

Example: This fixes #5012

**Describe the solution you've chosen**

Example: I've fixed this by [...]

**Describe alternatives you've considered**

Example: Another solution would be [...]

**Checklist**

Please add a `x` inside each checkbox:

- [ ] I have read the [contribution guidelines](../blob/main/CONTRIBUTING.md).
- [ ] The status checks are successful (continuous integration). Those can be seen below.

**A picture of a cute animal (not mandatory but encouraged)**
cute animal

### CODE REVIEW

File: src/downloader.ts [hunk 1]
```diff
@@ -31,17 +33,26 @@ const download = async (targetDirectory: string, versionRange: string) => {
...
-  const url = getReleaseURL(version)
-  const res = await fetch(url)
-
-  if (res.body === null) {
-    throw new Error('Could not download Deno')
-  }
-
-  return res.body
-}
+const downloadVersion = async (versionRange: string, logger: Logger) =>
```
Comment (eduardoboucas) [inline src/downloader.ts:36]: Totally not a blocker, but personally I think it'd be nicer to keep the retry logic separate from the download logic. You could keep `downloadVersion` as is, and then have `downloadVersionWithRetry` which just something like `pRetry(downloadVersion, { ... })`.

Up to you though!

**Review by eduardoboucas — DISMISSED** [review]

File: src/downloader.ts (hunk 1 above)
Comment (jackiewmacharia) [reply src/downloader.ts]: Changed it, which also reset the review.

**Review by eduardoboucas — DISMISSED** [review]
Nice!

**Review by eduardoboucas — APPROVED** [review]
✨

**Pull request merged**